nautobot/extras/models/models.py

```diff
@@ -112,6 +112,7 @@ class ConfigContext(BaseModel, ChangeLoggedModel, ConfigContextSchemaValidationM
     tenant_groups = models.ManyToManyField(to="tenancy.TenantGroup", related_name="+", blank=True)
     tenants = models.ManyToManyField(to="tenancy.Tenant", related_name="+", blank=True)
     tags = models.ManyToManyField(to="extras.Tag", related_name="+", blank=True)
+    dynamic_groups = models.ManyToManyField(to="extras.DynamicGroup", related_name="+", blank=True)
```

Comment:
We'll need a `limit_choices_to` here to make sure that only Device/VirtualMachine objects can be associated